flexmeasures/data/services/utils.py

```diff
@@ -0,0 +1,104 @@
+import functools
+from flask import current_app
+from rq.job import Job
+
+import hashlib
+import base64
+
+
+def make_hash_sha256(o):
+    """
+    SHA256 instead of Python's hash function because apparently, python native hashing function
+    yields different results on restarts.
+    Source: https://stackoverflow.com/a/42151923
+    """
+    hasher = hashlib.sha256()
+    hasher.update(repr(make_hashable(o)).encode())
+    return base64.b64encode(hasher.digest()).decode()
+
+
+def make_hashable(o):
+    """
+    Function to create hashes for dictionaries with nested objects
+    Source: https://stackoverflow.com/a/42151923
+    """
+    if isinstance(o, (tuple, list)):
+        return tuple((make_hashable(e) for e in o))
+
+    if isinstance(o, dict):
+        return tuple(sorted((k, make_hashable(v)) for k, v in o.items()))
+
+    if isinstance(o, (set, frozenset)):
+        return tuple(sorted(make_hashable(e) for e in o))
+
+    if callable(
+        getattr(o, "make_hashable", None)
+    ):  # checks if the object o has the method make_hashable
+        return o.make_hashable()
```

Comment:
Why not use [`o.__hash__()`](https://docs.python.org/3/glossary.html#term-hashable)?

Reply:
oh, even a (dis)advantage of this is that it works as a default for objects that we don't have a custom hash implementation.

I'll change `*.make_hashable` to` *.__hash__()` because, in case we want to avoid default hash implementations of certain objects, we can always (black/white)-list them.

Reply:
oh, I've tried this out and It turns out that `*.__hash__()` is being used in other places, for instance, to create sets (where objects are compared using their hash). 

For that reason, I would not override the default implementation. In addition to that,  we want to cherrypick which property changes make two objects differ. For instance, in the `Sensor` case, we don't consider changes on the annotations a reason to create a fresh job.